### PR TITLE
Upload and Copy now return the uids of new message instances.

### PR DESCRIPTION
**Pull request by gudmundurg74**
The response object returned from exec contains the uids but the functions were not exposing them. They are necessary to be able to create client side imap data entries without having to do selectMailbox+listMessages.

### CODE REVIEW

File: src/command-parser.js [hunk 1]
```diff
@@ -496,3 +496,30 @@ export function parseSEARCH (response) {
...
+
+  return value
+}
+
+/**
+ * Parses APPEND (upload) response.
+ * @param {Object} response
+ * @returns {String} The uid assigned to the uploaded message.
+ */
+export function parseASSIGN (response) {
```
Comment (nifgraup) [inline src/command-parser.js]: Shouldn't this be parseAPPEND?

Comment (nifgraup) [inline src/command-parser.js]: Shouldn't this be parseAPPEND?

File: README.md [hunk 2]
```diff
@@ -602,14 +602,38 @@ Where
...
+
+Resolves with the new uid of the message in the destination folder.
 
 Command: [COPY](http://tools.ietf.org/html/rfc3501#section-6.4.7)
 
 ### Example
 
 ```javascript
-client.copyMessages('INBOX', '1:5', '[Gmail]/Trash').then(() => { ... });
+client.copyMessages('INBOX', '1:5', '[Gmail]/Trash').then(({srcSeqSet, destSeqSet}) => { ... });
```
Comment (nifgraup) [inline README.md]: This should be about upload/APPEND, not COPY.

File: src/command-parser.js (hunk 1 above)
Comment (gudmundurg74) [reply src/command-parser.js]: yes!

File: README.md (hunk 2 above)
Comment (gudmundurg74) [reply README.md]: Right.

**Comment by gudmundurg74** [comment]
> The RFC https://tools.ietf.org/html/rfc3501 says "Responses: no specific responses for this command" for both COPY and APPEND commands. Will this work in general?

So these responses are defined in https://tools.ietf.org/html/rfc4315. I will update the parser functions and documentation to make sure they allow for missing data in the responses.

File: src/command-parser.js [hunk 3]
```diff
@@ -499,26 +499,26 @@ export function parseSEARCH (response) {
...
-  const value = { srcSeqSet: '', destSeqSet: '' }
   const copyuid = response && response.copyuid
-
   if (copyuid) {
-    value.srcSeqSet = copyuid[1]
-    value.destSeqSet = copyuid[2]
+    return {
+      srcSeqSet: copyuid[1],
+      destSeqSet: copyuid[2]
+    }
```
Comment (nifgraup) [inline src/command-parser.js]: Out of curiosity, what is in copyuid[0]?

Comment (gudmundurg74) [reply src/command-parser.js]: These are UIDVALIDITY numbers/codes, afaict, https://tools.ietf.org/html/rfc3501#section-7.1

File: README.md [hunk 4]
```diff
@@ -602,14 +602,38 @@ Where
...
+Where
+
+  * **destination** is the destination folder path. Example: '[Gmail]/Trash'
+  * **message** is the message to be uploaded
+  * **options** is an optional options object
+    * **flags** is an array of flags you want to set on the uploaded message. Defaults to [\Seen]. (optional)
+
+Resolves with the new uid of the message in the destination folder if the server supports [UIDPLUS](https://tools.ietf.org/html/rfc4315).
 
 Command: [COPY](http://tools.ietf.org/html/rfc3501#section-6.4.7)
```
Comment (nifgraup) [inline README.md]: APPEND, not COPY

File: src/command-parser-unit.js [hunk 5]
```diff
@@ -475,3 +477,38 @@ describe('parseSEARCH', () => {
...
+    expect(parseCOPY({})).to.equal(undefined)
+  })
+
+  it('should return undefined when response is not defined', () => {
+    expect(parseCOPY()).to.equal(undefined)
+  })
+})
+
+describe('parseAPPEND', () => {
+  it('should parse ASSIGN response', () => {
```
Comment (nifgraup) [inline src/command-parser-unit.js]: APPEND

**Pull request merged**